monitoring/mongodb/alerts.yaml

```diff
@@ -97,20 +103,65 @@ groups:
   - alert: ReplicationLagWarning
     expr: |
       label_replace(
-        max(
-            max(mongodb_rs_members_optimeDate{namespace="${namespace}",pod=~"${service}.*",member_state="PRIMARY"})
-                by(pod, rs_nm)
-            - ignoring(member_idx) group_right
-            min(mongodb_rs_members_optimeDate{namespace="${namespace}",pod=~"${service}.*",member_state="SECONDARY"})
-                by(pod, rs_nm, member_idx)
-        ) by(member_idx, rs_nm) / 1000 > 30
-      , "member_idx", "$1", "member_idx", "(${service}[^.]*)\\.${service}.*")
-    for: 10m
+        max by(rs_nm, member_idx) (
+          (
+            max by(pod, rs_nm) (
+              mongodb_rs_members_optimeDate{namespace="${namespace}",pod=~"${service}.*",member_state="PRIMARY"}
+            )
+            - ignoring(member_idx) group_right min by(pod, rs_nm, member_idx) (
+              mongodb_rs_members_optimeDate{namespace="${namespace}",pod=~"${service}.*",member_state="SECONDARY"}
+            )
```

Comment:
overall, this analysis is done on each pod : we compare the primary's oplog date vs all the secondary oplog date, as viewed by this pod.
→ this gives the max optime diff as viewed by this pod.

should it not be other way, i.e. for each member_idx, diff the max(PRIMARY) with the max(SECONDARY) [ignoring pods], which gives a simpler expression (no ignoring, no need to perfrom 2 nested max) and more precisely mesures the lag of that member_idx ?

Reply:
I'll check that approach yes (I'm also trying that on multi shard clusters to ensure it's future proof)